### PR TITLE
Optimize some code

### PR DESCRIPTION
**Pull request by echooymxq**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/AgentIDDecorator.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public AgentIDDecorator() {
                     Attributes mainAttribs = manifest.getMainAttributes();
                     String projectName = mainAttribs.getValue("Implementation-Vendor-Id");
                     if (projectName != null) {
-                        if (projectName.equals("org.apache.skywalking")) {
+                        if ("org.apache.skywalking".equals(projectName)) {
```
Comment (kezhenxu94) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/AgentIDDecorator.java]: `projectName` won't be null

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/RunningMode.java [hunk 2]
```diff
@@ -43,14 +43,14 @@ public static void setMode(String mode) {
      * @return true if in this status
      */
     public static boolean isInitMode() {
-        return MODE.equals("init");
+        return "init".equals(MODE);
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/RunningMode.java]: `MODE` won't be null

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/RunningMode.java [hunk 3]
```diff
@@ -43,14 +43,14 @@ public static void setMode(String mode) {
...
     }
 
     /**
      * No-init mode, the oap just starts up, but wouldn't do storage init.
      * @return true if in this status.
      */
     public static boolean isNoInitMode() {
-        return MODE.equals("no-init");
+        return "no-init".equals(MODE);
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/RunningMode.java]: Same here

**Comment by wu-sheng** [comment]
I am OK with PR, even it seems not necessary, at least, it is harmless either. 

@JaredTan95 As @kezhenxu94 left questions, please hold the merge before they are discussed.

**Comment by wu-sheng** [comment]
We don't force the 2 approvers, because I want the flexible and trust all committers.

**Pull request merged**